ast/ddl.go

```diff
@@ -675,7 +677,76 @@ type CreateTableStmt struct {
 
 // Restore implements Node interface.
 func (n *CreateTableStmt) Restore(ctx *RestoreCtx) error {
-	return errors.New("Not implemented")
+	ctx.WriteKeyWord("CREATE TABLE ")
+	if n.IfNotExists {
+		ctx.WriteKeyWord("IF NOT EXISTS ")
+	}
+
+	if err := n.Table.Restore(ctx); err != nil {
+		return errors.Annotate(err, "An error occurred while splicing CreateTableStmt Table")
+	}
+	ctx.WritePlain(" ")
+	if n.ReferTable != nil {
+		ctx.WriteKeyWord("LIKE ")
+		if err := n.ReferTable.Restore(ctx); err != nil {
+			return errors.Annotate(err, "An error occurred while splicing CreateTableStmt ReferTable")
+		}
+	}
+
+	if lenCols := len(n.Cols); lenCols > 0 {
+		ctx.WritePlain("(")
+		for i, col := range n.Cols {
+			if i > 0 {
+				ctx.WritePlain(",")
+			}
+			if err := col.Restore(ctx); err != nil {
+				return errors.Annotatef(err, "An error occurred while splicing CreateTableStmt ColumnDef: [%v]", i)
+			}
+		}
+		for i, constraint := range n.Constraints {
+			if i > 0 || lenCols >= 1 {
+				ctx.WritePlain(",")
+			}
+			if err := constraint.Restore(ctx); err != nil {
+				return errors.Annotatef(err, "An error occurred while splicing CreateTableStmt Constraints: [%v]", i)
+			}
+		}
+		ctx.WritePlain(")")
+	}
+
+	for i, option := range n.Options {
+		ctx.WritePlain(" ")
+		if err := option.Restore(ctx); err != nil {
+			return errors.Annotatef(err, "An error occurred while splicing CreateTableStmt TableOption: [%v]", i)
+		}
+	}
+
+	if n.Partition != nil {
+		ctx.WritePlain(" ")
+		if err := n.Partition.Restore(ctx); err != nil {
+			return errors.Annotate(err, "An error occurred while splicing CreateTableStmt Partition")
+		}
+	}
+
+	if n.Select != nil {
+		ctx.WritePlain(" ")
+
+		switch n.OnDuplicate {
+		case OnDuplicateCreateTableSelectError:
+			// nothing output
+		case OnDuplicateCreateTableSelectIgnore:
+			ctx.WriteKeyWord("IGNORE ")
+		case OnDuplicateCreateTableSelectReplace:
+			ctx.WriteKeyWord("REPLACE ")
+		}
+
+		ctx.WriteKeyWord("AS ")
```

Comment:
We could remove line 732 with

```go
switch n.OnDuplicate {
case OnDuplicateCreateTableSelectIgnore:
    ctx.WriteKeyWord(" IGNORE")
...
}
ctx.WriteKeyWord(" AS ")
```

The `switch` could be extracted to a `func (ty OnDuplicateCreateTableSelectType) String() string` function if you like.

Reply:
ok